tests/test_constrained.py

```diff
@@ -1225,6 +1225,44 @@ def test_functional_soft(self):
             self.assertIsInstance(new._soft[label].weight, numbers.Number)
             self.assertEqual(info.penalty, new._soft[label].penalty)
 
+    def test_functional_tuple_label(self):
+        cqm = dimod.ConstrainedQuadraticModel()
+
+        dimod_vars = [
+            dimod.Integer(i, lower_bound=-5000, upper_bound=5000) for i in range(63)
+        ]
+        num_discrete = 3
+        num_cases = 7
+        dimod_vars.extend(
+            [
+                dimod.Binary((c, i))
```

Comment:
Perhaps also add a case with nested tuples, and/or other collection types, like: `v = [{(1,(2,3)),4},5]`.